lms/services/lti_h.py

```diff
@@ -91,33 +91,17 @@ def upsert_course_group(self):
         isn't allowed to create groups (e.g. if they're just a student) then
         show an error page instead of continuing with the LTI launch.
         """
-
-        self._upsert_h_group(
-            group_id=self._context.h_groupid,
-            group_name=self._context.h_group_name,
-            creator=self._context.h_user,
-        )
-
-        self.group_info_service.upsert(
-            authority_provided_id=self._context.h_authority_provided_id,
-            consumer_key=self._request.lti_user.oauth_consumer_key,
-            params=self._request.params,
-        )
-
-    def _upsert_h_group(self, group_id, group_name, creator):
-        """Update the group and create it if the user is an instructor."""
+        group_id = self._context.h_groupid
+        group_name = self._context.h_group_name
 
         try:
             self.h_api.update_group(group_id=group_id, group_name=group_name)
-            return
-
         except HAPINotFoundError:
             # The group hasn't been created in h yet.
+            self.h_api.create_group(group_id=group_id, group_name=group_name)
 
-            if not self._request.lti_user.is_instructor:
-                raise HTTPBadRequest("Instructor must launch assignment first.")
-
-        # Try to create the group with the current instructor as its creator.
-        self.h_api.create_group(
-            group_id=group_id, group_name=group_name, creator=creator
+        self.group_info_service.upsert(
+            authority_provided_id=self._context.h_authority_provided_id,
+            consumer_key=self._request.lti_user.oauth_consumer_key,
+            params=self._request.params,
         )
```

Comment:
This method got refactored a bit, simplifying things by moving a separate private helper method into the main method. The only behaviour change is that it no longer passes `creator` to `create_group()`